tests/MediaTest.php

```diff
@@ -0,0 +1,340 @@
+use Illuminate\Http\Testing\File;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+use RonasIT\Support\Traits\FilesUploadTrait;
+
+class MediaTest extends TestCase
+{
+    use FilesUploadTrait;
+    use MediaTestTrait;
+
+    protected $admin;
+    protected $user;
+    protected File $file;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->admin = User::find(1);
+        $this->user = User::find(2);
+        $this->file = UploadedFile::fake()->image('file.png', 600, 600);
+    }
+
+    public function testCreate(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => $this->admin->id,
+            'is_public' => false,
+            'link' => '/storage/file.png',
+        ]);
+    }
+
+    public function testCreatePublic(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->user)->json('post', '/media', [
+            'file' => $this->file,
+            'is_public' => true,
+        ]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file.png',
+            'owner_id' => $this->user->id,
+            'is_public' => true,
+            'link' => '/storage/file.png',
+        ]);
+    }
+
+    public function testCreateCheckUrls(): void
+    {
+        $this->mockGenerateFilename();
+
+        $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $this->assertEquals(1, Media::where('link', 'like', '/%')->count());
+    }
+
+    public function testCreateCheckResponse(): void
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'link' => '/storage/file.png',
+        ]);
+
+        Storage::disk('local')->assertExists($this->getFilePathFromUrl('file.png'));
+
+        $this->clearUploadedFilesFolder();
+    }
+
+    public function testCreateNoAuth(): void
+    {
+        $response = $this->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertUnauthorized();
+    }
+
+    public function testBulkCreate(): void
+    {
+        /*
+        $this->mockGenerateFilename(2);
+        */
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs($this->admin)->json('post', '/media/bulk', [
+            'media' => [
+                [
+                    'file' => $this->file,
+                    'meta' => ['test1'],
+                ],
+                /*
```

Comment:
please keep two media in test for bulk creation

Reply:
Done.